src/textual/widgets/_text_area.py

```diff
@@ -0,0 +1,1857 @@
+from __future__ import annotations
+
+import re
+from collections import defaultdict
+from dataclasses import dataclass, field
+from functools import lru_cache
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Iterable, Optional, Tuple
+
+from rich.style import Style
+from rich.text import Text
+
+from textual._text_area_theme import TextAreaTheme
```

Comment:
Absolute imports again. Any reason for doing it that way? We've used relative elsewhere.